geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-api/src/main/scala/org/locationtech/geomesa/fs/storage/api/StorageMetadata.scala

```diff
@@ -53,6 +53,13 @@ trait StorageMetadata extends Compactable with Closeable {
     */
   def leafStorage: Boolean
 
+  /**
+    * The path filter factory used to filter paths during queries
+    *
+    * @return path filter factory
+    */
+  def pathFilterFactory: Option[PathFilterFactory]
```

Comment:
what do you think about moving this to a consructor arg in `ConverterStorage`, and then have it loaded in `ConverterStorageFactory`? then it could be instantiated in `ConverterStorage.createReader` and passed to `ConverterFileSystemReader`. that would avoid having to modify the StorageMetadata api, since it's only implemented for converter storage we could just isolate the changes there. i'm trying to think if this would be generally useful ever for other metadata impls, but i can't think of anything, since in those we control the file paths that are written.